plugins/inputs/azure_monitor/azure_monitor.go

```diff
@@ -0,0 +1,856 @@
+package azure_monitor
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type AzureMonitor struct {
+	azureClient *AzureClient
+
+	SubscriptionID       string                 `toml:"subscription_id"`
+	ClientID             string                 `toml:"client_id"`
+	ClientSecret         string                 `toml:"client_secret"`
+	TenantID             string                 `toml:"tenant_id"`
+	ResourceTargets      []*ResourceTarget      `toml:"resource_target"`
+	ResourceGroupTargets []*ResourceGroupTarget `toml:"resource_group_target"`
+	SubscriptionTargets  []*Resource            `toml:"subscription_target"`
+
+	Log telegraf.Logger `toml:"-"`
+}
+
+type ResourceTarget struct {
+	ResourceID  string   `toml:"resource_id"`
+	Metrics     []string `toml:"metrics"`
+	Aggregation []string `toml:"aggregation"`
+}
+
+type ResourceGroupTarget struct {
+	ResourceGroup string      `toml:"resource_group"`
+	Resources     []*Resource `toml:"resource"`
+}
+
+type Resource struct {
+	ResourceType string   `toml:"resource_type"`
+	Metrics      []string `toml:"metrics"`
+	Aggregation  []string `toml:"aggregation"`
+}
+
+type AzureClient struct {
+	client               *http.Client
+	accessToken          string
+	accessTokenExpiresOn time.Time
+}
+
+type Metric struct {
+	name   string
+	fields map[string]interface{}
+	tags   map[string]string
+}
+
+const (
+	maxMetricsPerRequest    int = 20
+	minMetricsFields            = 2
+	accessTokenURLGrantType     = "client_credentials"
+	accessTokenURLResource      = "https://management.azure.com/"
+)
+
+var sampleConfig = `
+# can be found under Overview->Essentials in the Azure portal for your application/service
+subscription_id = "<<SUBSCRIPTION_ID>>"
+# can be obtained by registering an application under Azure Active Directory
+client_id = "<<CLIENT_ID>>"
+# can be obtained by registering an application under Azure Active Directory
+client_secret = "<<CLIENT_SECRET>>"
+# can be found under Azure Active Directory->Properties
+tenant_id = "<<TENANT_ID>>"
```

Comment:
Those should be prefixed with two spaces to align with other plugins in the produced config.